sdk/create-dapp/templates/react-e2e-counter/src/Counter.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  useCurrentAccount,
+  useSignAndExecuteTransactionBlock,
+  useSuiClientQuery,
+} from "@mysten/dapp-kit";
+import { SuiObjectData } from "@mysten/sui.js/client";
+import { TransactionBlock } from "@mysten/sui.js/transactions";
+import { Button, Flex, Heading, Text } from "@radix-ui/themes";
+import { PACKAGE_ID } from "./constants";
+
+export function Counter({ id }: { id: string }) {
+  const currentAccount = useCurrentAccount();
+  const { mutateAsync: signAndExecute } = useSignAndExecuteTransactionBlock({});
+  const { data, isLoading, error, refetch } = useSuiClientQuery("getObject", {
+    id,
+    options: {
+      showContent: true,
+      showOwner: true,
+    },
+  });
+
+  const executeMoveCall = (method: "increment" | "reset") => {
+    const txb = new TransactionBlock();
+
+    if (method === "reset") {
+      txb.moveCall({
+        arguments: [txb.object(id), txb.pure.u64(0)],
+        target: `${PACKAGE_ID}::counter::set_value`,
+      });
+    } else {
+      txb.moveCall({
+        arguments: [txb.object(id)],
+        target: `${PACKAGE_ID}::counter::increment`,
+      });
+    }
+
+    signAndExecute({
+      requestType: "WaitForEffectsCert",
+      transactionBlock: txb,
+      options: {
+        showEffects: true,
+        showObjectChanges: true,
+      },
+    }).then(() => {
+      refetch();
```

Comment:
since the co-maintainer of react-query never uses refetch, I think I'd also prefer if we invalidated the query instead of hard refetching here 😆 

https://github.com/TanStack/query/discussions/2468